### ISSUE TITLE
[FR] Add a Video Player Plugin to AppFlowy Editor

### ISSUE THREAD

**Issue by annieappflowy** [AppFlowy-IO/AppFlowy#1067]
**1~3 main use cases of the proposed feature**
Enable users to insert videos into pages

**what types of users can benefit from using your proposed feature**
Users who use videos to express ideas

**Comment by MrHeer** [comment]
Can I take it?

**Comment by LucasXu0** [comment]
Sure. I assigned this issue to you.

**Comment by MrHeer** [comment]
Is there a more detailed requirements document and UI?

**Comment by LucasXu0** [comment]
At present, there is no perfect design draft. You can refer to the design of the image plugin, and support functionalities such as scaling, aligning, copying, deleting, etc.

**Comment by MrHeer** [comment]
Hi, I have no idea about video player. Do you have some suggestion? Should I develop a Flutter plugin?

**Comment by LucasXu0** [comment]
Hi, @MrHeer. You don't need to make a Flutter plugin. 
Here is the guiding document. [customizing-a-component](https://github.com/AppFlowy-IO/AppFlowy/blob/main/frontend/app_flowy/packages/appflowy_editor/documentation/customizing.md#customizing-a-component). Feel free to request more information if you have any questions after reading it.

**Comment by MrHeer** [comment]
But there should have a VideoNodeWidget like the ImageNodeWidget in example, In the example, use a Image.network to display a Image, but there are no weiget to render a video player.

**Comment by LucasXu0** [comment]
After doing a simple investigation, I haven't found such a widget to render video for the time being. Perhaps this feature can be suspended first.

**Comment by MrHeer** [comment]
Can I create a draft PR to waiting the video widget?

**Comment by LucasXu0** [comment]
Yup. I will keep looking for a cross-platform video player. (Web, Mobile and Desktop)

**Comment by ishtiaqshaheer** [comment]
Hi, is adding video player within editor available or ready yet?

**Comment by LucasXu0** [comment]
Bump it up. Finally, there's a [package](https://github.com/zezo357/flutter_meedu_videoplayer) that supports all platforms(not verified yet.).

**Comment by MrHeer** [comment]
I have a lot of time to continue to complete this task.

**Comment by zhouxinghong** [comment]
# markdown files: add support for online video and local video preview

Editing the markdown file in `AppFlowy` found that online video and local video preview cannot be supported, nor can YouTube video preview be supported. I hope that online video and local video preview can be supported, and streaming video preview such as YouTube can also be supported.

At present, all mainstream markdown editors provide more friendly and advanced video playback support. Only `AppFlowy` cannot provide complete video playback support. This completely limits the functional use of  `AppFlowy` in markdown editing.

Hopefully these video playback features will be supported in the next release.

Windows 10 19044
AppFlowy 0.3.9

**If image video information is present in the Markdown file:**
``` html
<video poster="https://cdn.up/iAgMxQoglUZ4uh7m4vFIhB5bLuljfjAtoBw8dOUV.jpg" src="https://ss201.dd.sg/video/hb.mp4">

</video>
```
**Then the image video information cannot be parsed in Markdown preview mode.**

**Comment by zhouxinghong** [comment]
Please support HTML5 online streaming video playback and image rendering.

Feature Request:
Please support inserting HTML5 code format for online streaming video playback and image rendering in Markdown files:

``` 
<video controls poster="https://ndn.up-timely.com/image/mP0qOkMCWtMW.jpg" src="https://z001.sbs.us/video/00580mhb.mp4">
</video
```

```
![](https://xdn.up-ti.com/71785/H5e4CWtMW.png)
```

Windows 10
AppFlowy 0.39


Almost all Markdown editors now support parsing the above code to insert and play HTML5 online streaming video and image rendering. Please support this feature as soon as possible. Thank you very much for your excellent MD editor. 🙏👍

**Comment by stickyfran** [comment]
Plz :( This and having thumbnail for uploaded videos

**Comment by bryantwilliam** [comment]
Hi just wondering how hard it'd be for me to do this? I'm not using ApplyFlowy yet.
